pandas/_libs/tslib.pyx

```diff
@@ -550,25 +536,13 @@ cpdef array_to_datetime(
 
                     string_to_dts_failed = string_to_dts(
```

Comment:
The error messaging here is a bit confusing to me - looks like `string_to_dts` is already labeled `?except -1`. Is there a reason why Cython doesn't propogate an error before your check of `if string_to_dts_failed`?

Reply:
It's because here `want_exc` is `False`:

https://github.com/pandas-dev/pandas/blob/a37b78d534cd25e21d2ab64318ceae808df46c89/pandas/_libs/tslibs/src/datetime/np_datetime_strings.c#L665-L671

The only place where it's True is

https://github.com/pandas-dev/pandas/blob/2f54a474aa754f37f82dab0184570e421207ba6d/pandas/_libs/tslib.pyx#L145-L162

which is only a testing function. So, perhaps the `?except -1` can just be removed, and the testing function removed (I think it would be better to test `to_datetime` directly)

Reply:
Cool thanks for review. Yea I'd be OK with your suggestion in a separate PR. Always good to clean this up - not sure we've handled consistently in the past